fondant/component.py

```diff
@@ -107,22 +117,19 @@ def _load_or_create_manifest(self) -> Manifest:
     def load(self, args: argparse.Namespace) -> dd.DataFrame:
         """Abstract method that loads the initial dataframe"""
 
-    def _process_dataset(self, dataset: FondantDataset) -> FondantDataset:
+    def _process_dataset(self, dataset: FondantDataset) -> dd.DataFrame:
         """This function takes in a FondantDataset object and processes the initial input dataframe
```

Comment:
I think some of the docstrings need to be updated here. Currently this is just used to load in the initial dataframe. Adding the index and subsets is currently handled by the `run` method